api/src/main/java/io/opentelemetry/api/trace/SpanContext.java

```diff
@@ -128,22 +104,21 @@ public void copyTraceFlagsHexTo(char[] dest, int destOffset) {
    *
    * @return the {@code TraceState} associated with this {@code SpanContext}.
    */
-  public abstract TraceState getTraceState();
+  TraceState getTraceState();
 
   /**
    * Returns {@code true} if this {@code SpanContext} is valid.
    *
    * @return {@code true} if this {@code SpanContext} is valid.
    */
-  @Memoized
-  public boolean isValid() {
-    return TraceId.isValid(getTraceIdHex()) && SpanId.isValid(getSpanIdHex());
+  default boolean isValid() {
+    return TraceId.isValid(getTraceIdAsHexString()) && SpanId.isValid(getSpanIdAsHexString());
   }
```

Comment:
this will become more expensive after this change, we should calculate this only once since it is used couple of times per instance created.

Reply:
Also, see above...this is still memoized by the implementation.